core/app/[locale]/(default)/product/[slug]/page.tsx

```diff
@@ -232,6 +233,8 @@ export default async function Product(props: Props) {
         incrementLabel={t('ProductDetails.increaseQuantity')}
         prefetch={true}
         product={getProduct(productPromise)}
+        productId={productId}
+        productTitle={productPromise.then((product) => product.name)}
```

Comment:
How come we have `productId` and `productTitle` props? Shouldn't the `product` prop have `id` and `name`/`title` that we can use?

Reply:
`product` is a promise here. Awaiting on `product` to get `productId` is conceptually backwards, IMHO, regardless of whether we have or don't have to await on it to get other props. `productTitle` resolution can be moved inside of `<ProductDetail/>`.

Reply:
Done.

Reply:
In practice, we want to stream in the whole product, though. We don't have to have to fetch the entire product just to pass the ID as that defeats the purpose of streamability, no?

It's not like the component fetches the product based on the `productId` prop